### ISSUE TITLE
Upstream `V = int` to protoc's EnumTypeWrapper runtime

### ISSUE THREAD

**Issue by nipunn1313** [dropbox/mypy-protobuf#169]
Currently, the `NewType('MyEnumValue', int)` only exists within mypy stubs, but not in the original python code generated by protoc.

This results in ugly callsites like
```
from proto.mylib import MyEnum

MYPY = False
if MYPY:
    from proto.mylib. import MyEnumValue
```

If we can upstream to protoc a simple line
```
MyEnumValue = int
```

Then, the callsites can use mypy-protobuf more smoothly by doing
```
from proto.mylib import MyEnum, MyEnumValue
```

**Comment by nipunn1313** [comment]
After talking to @ilevkivskyi - we found an alternate solution involving the metaclass feature which doesn't involve serious changes to upstream.

https://mypy-play.net/?mypy=0.790&python=3.9&gist=7d0f7d6cc46a98cf8ebfc071642ef825

Potentially, we could add a `V = int` to the EnumTypeWrapper class, but it wouldn't be needed for generated code.

**Comment by nipunn1313** [comment]
I requested adding `V = int` to
https://github.com/protocolbuffers/protobuf/pull/8182

**Comment by MHDante** [comment]
Hi @nipunn1313. I'd like to suggest reconsidering the approach taken to define enum member types through `MyEnum.V` (as much as I love the ergonomics). After investigating more into #211, I found that the current approach may be unfeasable for broader type checkers.

Please see the discussion here: microsoft/pyright#1602

In there, @erictraut suggests that the way MyEnum.V is implemented is untenable for single-pass type-checkers. While I understand that the current implementation functions well with mypy, It seems to rely on what could be argued as an illegal expression that mypy may not support in the future (as typing standards rigidify) 

I would suggest reverting this change, (as it is still quite new, and might not have had such a wide impact). I would also suggest blocking protocolbuffers/protobuf#8182 until consensus is reached.

I also have some ideas of alternative approaches to this that would not require changes to the protobuf repo, but I need some times to work them out. 
...

As an aside, It would be useful to add pyright as a secondary test that runs on this repo. Having 2 (or more) type-checkers can help illustrate areas where a particular type-checker's implementation obscures an error in typing. I can give this a shot (but perhaps in its own PR).

**Comment by erictraut** [comment]
I filed a separate but related issue here: https://github.com/dropbox/mypy-protobuf/issues/214

**Comment by nipunn1313** [comment]
Oops didn't loop back here. The issue mentioned by MHDante was resolved by #260 independently without compromising the ability to use `V`.

Still actively trying to get `.V` into the upstream protocolbuffers/protobuf#8182

**Comment by nipunn1313** [comment]
It landed upstream. Waiting for 3.20.0 to release and then will add (backward compatible) support here.
They used `ValueType = int` rather than `V = int`. I'll support both here for now, but update the docs to recommend `ValueType`
